File: StackWithArrays/stack.cpp
class Stack
{
public:
  int a[MAX];

  Stack(): top(-1) { }

  bool push(int x);
  int pop();
  int peek();
  bool isEmpty();

private:
  int top;
  const int MAX = 10;
};

bool Stack::push(int x)
{
  if (top >= (MAX - 1))
    return false;
  a[++top] = x;
  return true;
}

int Stack::pop()
{
  if (top < 0)
    return -1;
  int x = a[top--];
  return x;
}

int Stack::peek()
{
  if (top < 0)
    return 0;
  int x = a[top];
  return x;
}

bool Stack::isEmpty() { return top < 0; }

int main()
{
  Stack s;
  s.push(1);
  s.push(2);
  s.push(3);
  s.pop();
}

File: StackWithArrays/stack.cs
using System;

namespace ImplementStack
{
  class Stack
  {
    private int[] stack;
    private int top;
    private int max;

    public Stack(int size)
    {
      stack = new int[size];
      top = -1;
      max = size;
    }

    public void push(int item)
    {
      if (top == max - 1)
        return;
      stack[++top] = item;
    }

    public int pop()
    {
      if (top == -1)
        return -1;
      return stack[top--];
    }

    public int peek()
    {
      if (top == -1)
        return -1;
      return stack[top];
    }
  }

  class Program
  {
    static static void Main()
    {
      Stack p = new Stack(5);
      p.push(10);
      p.push(20);
      p.push(30);
      p.pop();
    }
  }
}
